src/DependencyInjection/CompilerPass/MiddlewarePass.php

```diff
@@ -114,8 +114,8 @@ private function registerMiddleware(ContainerBuilder $container, array $middlewa
             }
 
             if (!isset($options['handler'])) {
-                $handlerStack = new DefinitionDecorator('csa_guzzle.handler_stack');
-                $handlerStack->setClass('GuzzleHttp\HandlerStack');
+                $handlerStack = new Definition('csa_guzzle.handler_stack');
+                $handlerStack->setFactory('GuzzleHttp\HandlerStack::create');
```

Comment:
Instead of using `$handlerStack->setFactory('GuzzleHttp\HandlerStack::create')`..., could you rather use `$handlerStack->setFactory(['GuzzleHttp\HandlerStack', 'create'])`? Thanks! :)